### DISCUSSION TITLE
Does jmx_exporter data have to go through Categraf first?

### DISCUSSION THREAD

**Discussion by ZimaBlueee** [ccfos/nightingale#1880] [Q&A]
夜莺版本: 最新的**7.0**

我已查询官方文档/谷歌/百度, 都没有找到监控jvm相关搭建教程, 问题如下:

我从jmx_exporter提供的http请求中已拿到jvm和Tomcat的数据, 然后在夜莺的数据源中选择了Prometheus, 并填写了该http的url, 但是夜莺并没有显示出里面的数据, 并且报错json解析失败.

请问是需要部署Categraf来对接jmx_exporter , 然后用夜莺去对接Categraf吗?

还有一个问题 , 请问哪里可以找到仪表盘的模板呢? 比如jvm的仪表盘,  我是一个监控新人, 自己搭建的总觉得比不上专业的...

跪求大家回答, 十分感谢!

**Comment by UlricQin** [comment]
1. Java 程序在启动的时候通过 `-javaagent` 把 prometheus jmx 那个 jar 包引入，即可暴露 /metrics 数据
2. categraf 中配置 conf/input.prometheus/promtheus.toml 采集这个 /metrics 数据
3. 在夜莺的即时查询中就可以查询了，只要能查到数据就可以配置大盘了
